src/adlist.h

```diff
@@ -88,6 +88,13 @@ void listRewindTail(list *list, listIter *li);
 void listRotateTailToHead(list *list);
 void listRotateHeadToTail(list *list);
 void listJoin(list *l, list *o);
+static inline void listInitEmbeddedNode(listNode * node, void *parentStruct) {
+    node->prev = (void * )0;
+    node->next = (void * )0;
+    node->value = parentStruct;
+}
+void listAddExistingNodeHead(list *list, listNode *embeddedNode);
+void listRemoveNode(list *list, listNode *embeddedNode);
```

Comment:
It's kind of hard to differentiate the name `listRemoveNode` from `listDelNode`. Alternatively, maybe `listLinkHeadNode` and `listUnlinkNode` ? We are linking nodes into the list.

```suggestion
void listAddExistingNodeHead(list *list, listNode *node);
void listRemoveNode(list *list, listNode *node);
```

Following from previous comment about the embedded term.

Reply:
I like listLink and ListUnlik, will make the change in the next revision